From 1c4b8bb6c646a492f8991e7e6a4c8594143b75e4 Mon Sep 17 00:00:00 2001
From: Nael Masalha <nael.masalha@intel.com>
Date: Thu, 25 Nov 2021 14:03:48 +0200
Subject: [PATCH] Enable IMU streaming

 - This is based on HW short to add I2C device and expose video node.

Signed-off-by: Nael Masalha <nael.masalha@intel.com>
---
 common/tegra194-camera-d4xx.dtsi | 68 +++++++++++++++++++++++++++++---
 1 file changed, 63 insertions(+), 5 deletions(-)

diff --git a/common/tegra194-camera-d4xx.dtsi b/common/tegra194-camera-d4xx.dtsi
index 09edc49..500101d 100644
--- a/common/tegra194-camera-d4xx.dtsi
+++ b/common/tegra194-camera-d4xx.dtsi
@@ -182,6 +182,60 @@
 					};
 				};
 			};
+			i2c@3 {
+				reg = <0x5>;	// line 4 of the I2C switch
+				#address-cells = <0x1>;
+				i2c-mux,deselect-on-exit;
+				#size-cells = <0x0>;
+
+				d4m3: d4m@10 {
+					status = "ok";
+					reg = <0x10>;
+					compatible = "intel,d4xx";
+					vcc-supply = <&p2822_vdd_1v8_cvb>;
+					/*reset-gpios = <&tegra_main_gpio CAM0_RST_L GPIO_ACTIVE_HIG>;*/
+					cam-type = "IMU";
+					ports {
+						#address-cells = <1>;
+						#size-cells = <0>;
+
+						port@0 {
+							reg = <0>;
+							d4m3_out: endpoint {
+								port-index = <1>;
+								bus-width = <2>;
+								remote-endpoint = <&csi_in3>;
+								src_imu: source@0 {
+									data-types = <0x2a>;
+									vc-id = <3>;
+								};
+							};
+						};
+					};
+					mode0 {
+						pixel_t = "grey_y8";
+						num_lanes = "2";
+						active_w = "640";
+						active_h = "480";
+						tegra_sinterface = "serial_b";
+						mclk_khz = "24000";
+						/*pix_clk_hz = "148500000";*/ /* for 2 streams? */
+						/*pix_clk_hz = "37125000";*/
+						/*pix_clk_hz = "375000000";*/
+						pix_clk_hz = "74250000";
+						discontinuous_clk = "no";
+						dpcm_enable = "false";
+						cil_settletime = "0";
+						line_length = "1280"; /* 2200 */
+						mclk_multiplier = "15.625";
+						embedded_metadata_height = "0";
+						min_framerate = "5";
+						max_framerate = "30";
+						default_framerate = "30";
+						vc_id = "3";
+					};
+				};
+			};
 
 		};
 	};
@@ -284,7 +338,7 @@
 			};
 			channel@3 {
 				reg = <0x3>;
-				status = "disabled";
+				status = "ok";
 
 				ports {
 					#address-cells = <0x1>;
@@ -296,9 +350,9 @@
 
 						csi_in3: endpoint@6 {
 							status = "ok";
-							port-index = <3>;
+							port-index = <0>;
 							bus-width = <2>;
-							//remote-endpoint = <&d4m3_out>;
+							remote-endpoint = <&d4m3_out>;
 						};
 					};
 					port@1 {
@@ -416,12 +470,13 @@
 				};
 				port@3 {
 					reg = <3>;
-					status = "disabled";
+					status = "ok";
 
 					vi_in3: endpoint {
-						status = "disabled";
+						status = "ok";
 						port-index = <0>;
 						bus-width = <2>;
+						vc-id = <3>;
 						remote-endpoint = <&csi_out3>;
 					};
 				};
@@ -456,6 +511,9 @@
 				source@1 {
 					remote-source = <&src_y80>;
 				};
+				source@2 {
+					remote-source = <&src_imu>;
+				};
 				//source@2 {
 				//	remote-source = <&src_depth1>;
 				//};
-- 
2.17.1

